azure/client.go

```diff
@@ -4,35 +4,35 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-04-01/network"
-	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
-	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
 )
 
 type client struct {
-	rgClient       resources.GroupsClient
-	sgClient       network.ApplicationSecurityGroupsClient
-	autorestClient autorest.Client
+	rgClient armresources.ResourceGroupsClient
+	sgClient armnetwork.ApplicationSecurityGroupsClient
 }
 
-func NewClient(rg resources.GroupsClient, sg network.ApplicationSecurityGroupsClient) client {
+func NewClient(rg armresources.ResourceGroupsClient, sg armnetwork.ApplicationSecurityGroupsClient) client {
 	return client{
-		rgClient:       rg,
-		sgClient:       sg,
-		autorestClient: rg.Client,
+		rgClient: rg,
+		sgClient: sg,
 	}
 }
 
 func (c client) ListGroups() ([]string, error) {
 	ctx := context.Background()
 	groups := []string{}
 
-	for list, err := c.rgClient.ListComplete(ctx, "", nil); list.NotDone(); err = list.Next() {
+	pager := c.rgClient.NewListPager(nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
 		if err != nil {
-			return []string{}, fmt.Errorf("List Complete Resource Groups: %s", err)
+			return groups, fmt.Errorf("failed to advance page: %s", err)
```

Comment:
In this case, I'd continue to return an empty list of groups since it's not the complete result and also keep the error wrapping message as is to keep it consistent with the other iaas client err wrapping.

Reply:
your right. i will correct this